src/binarymanager.ts

```diff
@@ -0,0 +1,124 @@
+import axios, { AxiosResponse } from 'axios';
+import which from 'which';
+import { globalContext } from './extension';
+import { MirrordAPI } from './api';
+import { Utils } from 'vscode-uri';
+import * as fs from 'node:fs';
+import { Uri, workspace, window, ProgressLocation } from 'vscode';
+
+const mirrordBinaryEndpoint = 'https://version.mirrord.dev/v1/version';
+const binaryCheckInterval = 1000 * 60 * 3;
+
+
+function getExtensionMirrordPath(): Uri {
+    return Utils.joinPath(globalContext.globalStorageUri, 'mirrord');
+}
+
+
+/**
+ * Returns path to mirrord binary
+ */
+export async function getMirrordBinaryPath(): Promise<string> {
```

Comment:
The name is misleading because it implies it just gets the path, but it actually downloads the binary as well. Consider `getMirrordBinary`